### PR TITLE
GloriaLoi.12-rest

### PR DESCRIPTION
**Pull request by gloooor**

### CODE REVIEW

File: 12-rest/classworks/Gloria_Loi/index.js [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+        body: JSON.stringify(requestBody),
+        headers: {
+          'Content-Type': 'application/json',
+          'User-Agent': 'insomnia/2020.3.3',
+        },
+      });
+
+      const data = await fetchData.json();
+
+      const tagline = [
```
Comment (KuhArt) [inline 12-rest/classworks/Gloria_Loi/index.js]: Лучше хранить информацию в виде объекта. Например
```
{
author, // author name
text, // text result
}
```

File: 12-rest/classworks/Gloria_Loi/index.js [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+      ].join(' ');
+
+      taglines.push(tagline);
+      fs.writeFileSync(
+        './database/tagline.json',
+        JSON.stringify(taglines, null, 2)
+      );
+      if (response) {
+        response.body = JSON.stringify({
+          message: 'successfully posted',
```
Comment (KuhArt) [inline 12-rest/classworks/Gloria_Loi/index.js]: В ответе чаще всего отправляется созданный объект.

**Review by KuhArt — CHANGES_REQUESTED** [review]

File: 12-rest/classworks/Gloria_Loi/database/tagline.json [hunk 3]
```diff
@@ -0,0 +1,10 @@
+[
+  "AUTHOR NAME: 2 волк не лев шей-то посмотрел?",
```
Comment (mikitasolo) [inline 12-rest/classworks/Gloria_Loi/database/tagline.json]: но в цирке не выступает :point_up:

File: 12-rest/classworks/Gloria_Loi/index.js [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+const crypto = require('crypto');
+const fs = require('fs');
+const koaJwt = require('koa-jwt');
+const jwt = require('jsonwebtoken');
+const fetch = require('node-fetch');
+
+const public = Router();
+const private = Router();
+
+const mySecret = 'super-mega-ochen-secret';
```
Comment (mikitasolo) [inline 12-rest/classworks/Gloria_Loi/index.js:13]: i like it

File: 12-rest/classworks/Gloria_Loi/index.js [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+
+public.post(
+  '/signup',
+  async ({
+    request: {
+      body: { password, login },
+    },
+    response
+  }) => {
+    const pw = crypto
```
Comment (mikitasolo) [inline 12-rest/classworks/Gloria_Loi/index.js]: you use this func at least twice. it's a good point to make it a separate function

Comment (mikitasolo) [inline 12-rest/classworks/Gloria_Loi/index.js]: better name it passwordHash as it stores passwordHash

File: 12-rest/classworks/Gloria_Loi/index.js [hunk 6]
```diff
@@ -0,0 +1,211 @@
...
+      .pbkdf2Sync(password, 'salt', 100000, 64, 'sha512')
+      .toString('hex');
+    const user = {
+      login,
```
Comment (mikitasolo) [inline 12-rest/classworks/Gloria_Loi/index.js]: i think we should not allow signing up with the same logins

File: 12-rest/classworks/Gloria_Loi/index.js [hunk 7]
```diff
@@ -0,0 +1,211 @@
...
+const jwt = require('jsonwebtoken');
+const fetch = require('node-fetch');
+
+const public = Router();
+const private = Router();
+
+const mySecret = 'super-mega-ochen-secret';
+
+const users = JSON.parse(fs.readFileSync('./database/file.json'));
+let user;
```
Comment (mikitasolo) [inline 12-rest/classworks/Gloria_Loi/index.js]: it should not be a global variable. imagine someone is logging in and at this moment another user trie to log in and can rewrite this var. better create a user in every log in request

File: 12-rest/classworks/Gloria_Loi/index.js [hunk 8]
```diff
@@ -0,0 +1,211 @@
...
+        body: JSON.stringify(requestBody),
+        headers: {
+          'User-Agent': 'insomnia/2020.3.3',
+          'Content-Type': 'application/json',
+        },
+      });
+
+      const res = await fetchData.json();
+      const tag = {
+        autorName: user.login,
```
Comment (mikitasolo) [inline 12-rest/classworks/Gloria_Loi/index.js]: better replace by `author`

File: 12-rest/classworks/Gloria_Loi/index.js [hunk 9]
```diff
@@ -0,0 +1,211 @@
...
+      },
+      response,
+    }) => {
+      const taglines = JSON.parse(fs.readFileSync('./database/tagline.json'));
+      const requestBody = {
+        prompt: combination,
+        length: 10,
+        num_samples: 1,
+      };
+      const fetchData = await fetch('https://models.dobro.ai/gpt2/medium/', {
```
Comment (mikitasolo) [inline 12-rest/classworks/Gloria_Loi/index.js]: it would be nice if you make it a separate function as you write it multiple times

**Review by mikitasolo — CHANGES_REQUESTED** [review]
Good job, I like it in general!

**Review by mikitasolo — APPROVED** [review]

**Review by mikitasolo — APPROVED** [review]

**Pull request merged**